modules/pricer/src/test/java/com/opengamma/strata/pricer/swap/DiscountingSwapProductPricerTest.java

```diff
@@ -747,4 +751,35 @@ public void par_spread_sensitivity_ibor_ibor() {
     assertTrue(prAd.equalWithTolerance(prFd, TOLERANCE_RATE_DELTA));
   }
 
+  //-------------------------------------------------------------------------
+  public void three_leg_swap() {
+    ThreeLegBasisSwapConvention conv = ThreeLegBasisSwapConventions.EUR_FIXED_1Y_EURIBOR_3M_EURIBOR_6M;
+    LocalDate tradeDate = LocalDate.of(2014, 1, 22);
+    Swap swap = conv.toTrade(tradeDate, Period.ofMonths(1), TENOR_5Y, BUY, NOTIONAL_SWAP, SPREAD).getProduct();
+    // pv
+    MultiCurrencyAmount pvComputed = SWAP_PRODUCT_PRICER.presentValue(swap, MULTI_EUR);
+    DiscountingSwapLegPricer legPricer = SWAP_PRODUCT_PRICER.getLegPricer();
+    CurrencyAmount pvExpected = legPricer.presentValue(swap.getLegs().get(0), MULTI_EUR)
+        .plus(legPricer.presentValue(swap.getLegs().get(1), MULTI_EUR))
+        .plus(legPricer.presentValue(swap.getLegs().get(2), MULTI_EUR));
+    assertEquals(pvComputed.getAmount(EUR), pvExpected);
+    // pv sensitivity
+    PointSensitivityBuilder pvPointComputed = SWAP_PRODUCT_PRICER.presentValueSensitivity(swap, MULTI_EUR);
+    PointSensitivityBuilder pvPointExpected = legPricer.presentValueSensitivity(swap.getLegs().get(0), MULTI_EUR)
+        .combinedWith(legPricer.presentValueSensitivity(swap.getLegs().get(1), MULTI_EUR))
+        .combinedWith(legPricer.presentValueSensitivity(swap.getLegs().get(2), MULTI_EUR));
+    assertEquals(pvPointComputed, pvPointExpected);
+    // par rate
+    double parRate = SWAP_PRODUCT_PRICER.parRate(swap, MULTI_EUR);
+    Swap swapPar = conv.toTrade(tradeDate, Period.ofMonths(1), TENOR_5Y, BUY, NOTIONAL_SWAP, parRate).getProduct();
+    MultiCurrencyAmount pvPar = SWAP_PRODUCT_PRICER.presentValue(swapPar, MULTI_EUR);
+    assertEquals(pvPar, MultiCurrencyAmount.of(EUR, 0d));
+    // par rate sensitivity
+    PointSensitivities parPoint = SWAP_PRODUCT_PRICER.parRateSensitivity(swap, MULTI_EUR).build();
+    CurveCurrencyParameterSensitivities parSensiComputed = MULTI_EUR.curveParameterSensitivity(parPoint);
+    CurveCurrencyParameterSensitivities parSensiExpected = FINITE_DIFFERENCE_CALCULATOR.sensitivity(MULTI_EUR,
+        p -> CurrencyAmount.of(EUR, SWAP_PRODUCT_PRICER.parRate(swap, p)));
+    assertTrue(parSensiComputed.equalWithTolerance(parSensiExpected, TOLERANCE_RATE_DELTA));
+  }
```

Comment:
Should check parSpread and parSpreadSensitivity as it will be used for curve calibration.